src/pages/MovieListPage/MovieListPage.tsx

```diff
@@ -0,0 +1,158 @@
+        setFilters((prevState) => ({
+            ...prevState,
+            [key] : value,
+        }));
+    }
+
+    function handleSearch(searchInput: string) {
+        updateFilters("searchQuery", searchInput);
+    }
+
+    function handleGenreSelect(genre: string) {
+        updateFilters("selectedGenre", genre === "All" ? "" : genre);
+    }
+
+    function handleSortControlChange(option: string) {
+        updateFilters("selectedSortControl", option);
+    }
+
+    function handleTileClick(movieDetails: MovieDetailsData) {
+        setSelectedMovie(movieDetails);
+    }
+
+    function handleToggleDialog() {
+        setShowDialog(!showDialog);
+    }
+
+    function handleShowDialog(dialogOption: string, movieDetails?: MovieDetailsData) {
+        setCurrentDialog(dialogOption);
+
+        if (movieDetails) {
+            setSelectedMovie(movieDetails);
+        }
+
+        handleToggleDialog();
+    }
+
+    function handleSubmit(MovieData: MovieDetailsData) {
+        console.log(MovieData);
+    }
+
+    function handleBackToSearch() {
+        setSelectedMovie(null);
+    }
+
+    return (
+        <div className={
+            showDialog
+                ? styles.movieListPageContainer + " " + styles.movieListPageContainerBlured
+                : styles.movieListPageContainer
+            }
+        >
+
+            {selectedMovie ? (
+                <MovieDetails movieDetails={selectedMovie} handleBackToSearch={handleBackToSearch}/>
+            ) : (
+                <>
+                    <SearchForm initialSearch={"What do you want to watch?"}
+                                onSearch={handleSearch}/>
+                    <AddMovieButton handleShowDialog={handleShowDialog} />
+                </>
+            )}
+
+            <div className={styles.genreAndSortControls}>
+                <GenreSort genreNames={GenreNames}
+                           selectedGenre={filters.selectedGenre}
+                           onSelect={handleGenreSelect}/>
+                <SortControl currentSelection={filters.selectedSortControl}
+                             onSelect={handleSortControlChange}/>
+            </div>
+
+            {isLoading ? <p>Loading movies...</p> :
+                <MovieList movieList={movieList}
+                           handleTileClick={handleTileClick}
+                           handleEditMovie={handleShowDialog}
+                           handleDeleteMovie={handleShowDialog}
+                />
+            }
+            {(!isLoading && movieList.length === 0) && <p>No movies found</p>}
+
+            <Dialog dialogTitle={currentDialog}
+                    content={<ModalContent currentDialog={currentDialog}
+                                           selectedMovie={selectedMovie}
+                                           handleSubmit={handleSubmit} />}
+                    handleToggleDialog={handleToggleDialog}
+                    showDialog={showDialog}/>
+        </div>
```

Comment:
As usual Page doesn't have include so many logic, functions, html code. Would be better to create Header, Main, and Footer. Try to separate to different components/containers. Do not forget to think about how to manage state.